pkg/kv/kvserver/spanset/engine.go

```diff
@@ -0,0 +1,289 @@
+	return s.e.Attrs()
+}
+
+// Capacity implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) Capacity() (roachpb.StoreCapacity, error) {
+	return s.e.Capacity()
+}
+
+// Properties implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) Properties() roachpb.StoreProperties {
+	return s.e.Properties()
+}
+
+// ProfileSeparatedValueRetrievals implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) ProfileSeparatedValueRetrievals(
+	ctx context.Context,
+) (*metrics.ValueRetrievalProfile, error) {
+	return s.e.ProfileSeparatedValueRetrievals(ctx)
+}
+
+// Compact implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) Compact(ctx context.Context) error {
+	return s.e.Compact(ctx)
+}
+
+// Env implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) Env() *fs.Env {
+	return s.e.Env()
+}
+
+// Flush implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) Flush() error {
+	return s.e.Flush()
+}
+
+// GetMetrics implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) GetMetrics() storage.Metrics {
+	return s.e.GetMetrics()
+}
+
+// GetEncryptionRegistries implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) GetEncryptionRegistries() (*fs.EncryptionRegistries, error) {
+	return s.e.GetEncryptionRegistries()
+}
+
+// GetEnvStats implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) GetEnvStats() (*fs.EnvStats, error) {
+	return s.e.GetEnvStats()
+}
+
+// GetAuxiliaryDir implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) GetAuxiliaryDir() string {
+	return s.e.GetAuxiliaryDir()
+}
+
+// IngestLocalFiles implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) IngestLocalFiles(ctx context.Context, paths []string) error {
+	return s.e.IngestLocalFiles(ctx, paths)
+}
+
+// IngestLocalFilesWithStats implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) IngestLocalFilesWithStats(
+	ctx context.Context, paths []string,
+) (pebble.IngestOperationStats, error) {
+	return s.e.IngestLocalFilesWithStats(ctx, paths)
+}
+
+// IngestAndExciseFiles implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) IngestAndExciseFiles(
+	ctx context.Context,
+	paths []string,
+	shared []pebble.SharedSSTMeta,
+	external []pebble.ExternalFile,
+	exciseSpan roachpb.Span,
+) (pebble.IngestOperationStats, error) {
+	return s.e.IngestAndExciseFiles(ctx, paths, shared, external, exciseSpan)
+}
+
+// IngestExternalFiles implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) IngestExternalFiles(
+	ctx context.Context, external []pebble.ExternalFile,
+) (pebble.IngestOperationStats, error) {
+	return s.e.IngestExternalFiles(ctx, external)
+}
+
+// IngestLocalFilesToWriter implements the storage.EngineWithoutRW interface.
+func (s *spanSetEngine) IngestLocalFilesToWriter(
```

Comment:
We probably can't check much for the ingestions, but some things we can like `exciseSpan` and `clearedSpans` here. Do?

Reply:
Actually, `clearedSpans` will itself be written to a batch, which will already assert. So maybe skip this one. But the `exciseSpan` seems assertable.